MISC: Export Azure DevOps Build Pipeline config

nielsboecker opened the pull request:
This PR will extract our CI config.

nielsboecker committed:
MISC: Add azure-pipelines.yml configuration

nielsboecker committed:
MISC: Add legacy JSON config

nielsboecker committed:
MISC: Add DevOps build pipeline README

nbckr commented:
Imported it as a new pipeline to demonstrate it works: https://dev.azure.com/MSGOSHHOLO/HoloRepository/_build?definitionId=5

The new pipeline setting can be deleted once this PR is closed.

nielsboecker committed:
MISC: Fix training whitespaces

boonwj committed:
Add comment for REACT_APP_BACKEND_HOST

Added a note to explain why the REACT_APP_BACKEND_HOST is hardcoded into the builds.

boonwj committed:
MISC: Fix trailing space errors

Interestingly there are still trailing space errors

boonwj committed:
MISC: Remove stray link in README

boonwj committed:
MISC: Removed legacy pipeline configurations

Remove need to maintain 2 separate build pipeline files.

boonwj commented:
It's weird that CodeFactor still finds errors. The pipelines are working as expected.
